### PR TITLE
Develop to Main

### PR DESCRIPTION
**Pull request by menachem554**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DATABASE_URL=mongodb://localhost/scholl
```
Comment (Shahar-Y) [inline .env]: The `.env` file should be in gitignore. remove it from the code.

File: src/controller/students.controller.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
+/* eslint-disable prefer-const */
+/* eslint-disable import/no-unresolved */
```
Comment (Shahar-Y) [inline src/controller/students.controller.ts]: remove rules from all of the files except eslintrc

File: src/controller/students.controller.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import { ITacher } from '../interface/tacher.interface';
+import StudentSchema from '../models/student.model';
+import TacherSchema from '../models/tacher.model';
+import * as valdiat from './tacher.controller';
+
+const router = express.Router();
+
+// Add Student
+export const postStudent = async (req: Request, res: Response) => {
+  const student = new StudentSchema({
```
Comment (Shahar-Y) [inline src/controller/students.controller.ts]: The schema shpould be in the repository, not in the controller. same for all functions

File: src/controller/students.controller.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    // get all body Parameters
+    sName: req.body.sName,
+    age: req.body.age,
+    averageGrade: req.body.averageGrade,
+    studentID: req.body.studentID,
+  });
+
+  // Save the new student
+  try {
+    const newStudent = await student.save();
```
Comment (Shahar-Y) [inline src/controller/students.controller.ts]: what is the type of `newStudent`?

File: src/controller/students.controller.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+  } catch (err: any) {
+    res.status(500).json({ message: err.message });
+  }
+};
+
+// Get student profession (no Query Params here)
+export const getStudentProfession = async (req: Request, res: Response) => {
+  try {
+    // Aggregate the students profession from tachers collection
+    const student: IStudent[] = await StudentSchema.aggregate([
```
Comment (Shahar-Y) [inline src/controller/students.controller.ts]: same as before. this aggregation should be in the repository

File: src/controller/tacher.controller.ts [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+      tempResult = calculat / sum;
+      calculat = 0;
+
+      if (result < tempResult) {
+        result = tempResult;
+        tacherID = doc.tacherID;
+        tacherName = doc.tName;
+      }
+    });
```
Comment (Shahar-Y) [inline src/controller/tacher.controller.ts]: This calculation can be in an auxiliary function

File: src/controller/tacher.controller.ts [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+    console.log();
```
Comment (Shahar-Y) [inline src/controller/tacher.controller.ts]: ?

File: src/controller/tacher.controller.ts [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    },
+  ]);
+
+  tacher.forEach(async (doc) => {
+    studentList = doc.studentList;
+    tacherIDs = doc.tacherID;
+
+    for (let i: number = 0; i < studentList.length; i += 1) {
+      if (studentID == studentList[i]) {
+        // eslint-disable-next-line no-await-in-loop
```
Comment (Shahar-Y) [inline src/controller/tacher.controller.ts]: remove rule

File: src/models/tacher.model.ts [hunk 9]
```diff
@@ -1,8 +1,8 @@
-/* eslint-disable prettier/prettier */
+/* eslint-disable import/no-unresolved */
```
Comment (Shahar-Y) [inline src/models/tacher.model.ts]: remove rule

File: src/routes/student.router.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable import/no-unresolved */
```
Comment (Shahar-Y) [inline src/routes/student.router.ts]: remove rule

File: src/routes/tacher.router.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable import/no-unresolved */
```
Comment (Shahar-Y) [inline src/routes/tacher.router.ts]: remove rule

File: src/server.ts [hunk 12]
```diff
@@ -1,27 +1,29 @@
-/* eslint-disable prettier/prettier */
-
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-unresolved */
```
Comment (Shahar-Y) [inline src/server.ts]: remove rules

File: src/interface/tacher.interface.ts [hunk 13]
```diff
@@ -1,12 +1,16 @@
-/* eslint-disable prettier/prettier */
-// eslint-disable-next-line import/no-unresolved
-import  {ProfessionEnum}  from './student.interface';
-
-// Tachers collection:  will contain a list of tachers
+/* eslint-disable no-shadow */
```
Comment (Shahar-Y) [inline src/interface/tacher.interface.ts]: remove rule

**Review by Shahar-Y — CHANGES_REQUESTED** [review]
Good job. seperate the controller and the repository

File: .env (hunk 1 above)
Comment (menachem554) [reply .env]: done

File: src/controller/students.controller.ts (hunk 2 above)
Comment (menachem554) [reply src/controller/students.controller.ts]: done

File: src/controller/students.controller.ts (hunk 3 above)
Comment (menachem554) [reply src/controller/students.controller.ts]: done

File: src/controller/students.controller.ts (hunk 4 above)
Comment (menachem554) [reply src/controller/students.controller.ts]: done

File: src/controller/students.controller.ts (hunk 5 above)
Comment (menachem554) [reply src/controller/students.controller.ts]: done

File: src/models/tacher.model.ts (hunk 9 above)
Comment (menachem554) [reply src/models/tacher.model.ts]: done

File: src/routes/student.router.ts (hunk 10 above)
Comment (menachem554) [reply src/routes/student.router.ts]: done

File: src/routes/tacher.router.ts (hunk 11 above)
Comment (menachem554) [reply src/routes/tacher.router.ts]: done

File: src/server.ts (hunk 12 above)
Comment (menachem554) [reply src/server.ts]: done

File: src/interface/tacher.interface.ts (hunk 13 above)
Comment (menachem554) [reply src/interface/tacher.interface.ts]: done

File: src/managers/tacher.manager.ts [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+  const tacher: ITacher[] = await tacherRepo.tachersList();
+
+  tacher.forEach(async (doc) => {
+    studentList = doc.studentList;
+    tacherIDs = doc.tacherID;
+
+    for (let i: number = 0; i < studentList.length; i += 1) {
+      if (studentID === studentList[i]) {
+        const del = tacherRepo.delSsntFromTchr(tacherIDs, studentList[i]);
+        console.log(del);
```
Comment (Shahar-Y) [inline src/managers/tacher.manager.ts:132]: remove all redundant and meaningless console.logs from the code

**Review by Shahar-Y — APPROVED** [review]
Well done. Approved!

**Comment by Shahar-Y** [comment]
Notice you have many conflicts with the main branch!

**Pull request merged**